libraries/package-extractor/src/PackageExtractor.ts

```diff
@@ -483,8 +515,15 @@ export class PackageExtractor {
           path.join(packageJsonRealFolderPath, 'package.json')
         );
 
+        const targetSubspace: IExtractorSubspace | undefined = subspaces?.find(
+          (p) => p.pnpmInstallFolder && Path.isUnder(packageJsonFolderPath, p.pnpmInstallFolder)
+        );
+
         // Transform packageJson using the provided transformer, if requested
-        const packageJson: IPackageJson = transformPackageJson?.(originalPackageJson) ?? originalPackageJson;
+        const packageJson: IPackageJson =
+          targetSubspace?.transformPackageJson?.(originalPackageJson) ??
+          transformPackageJson?.(originalPackageJson) ??
+          originalPackageJson;
```

Comment:
We should validate that only one of the two options (`subspaces` vs top-level `transformPackageJson`/`pnpmInstallFolder`) is provided, and throw an error if it's not the case. We wouldn't want to hide why the top-level ones are ignored when someone specified `subspaces`

Reply:
Fixed in my https://github.com/EscapeB/rushstack/pull/1